lib/dialogs/settings/panels/tools.tsx

```diff
@@ -43,9 +48,8 @@ ToolsPanel.propTypes = {
 };
 
 const mapDispatchToProps = dispatch => {
-  const { showDialog } = appState.actionCreators;
   return {
-    showImportDialog: () => dispatch(showDialog({ dialog: IMPORT })),
+    showImportDialog: () => dispatch(showDialog('IMPORT')),
```

Comment:
we're in a type-mismatch here that doesn't show because we haven't added a type signature to `mapDispatchToProps`.

If we were to do that…

```ts
const mapDispatchToProps: S.MapDispatch<DispatchProps> = …
```

then the compiler would warn us that we're not sending `showDialog`

We can either update `ToolsPanel` to call `showDialog` and update this reference in `mapDispatchToProps` or we can rename the type in `DispatchProps` to `showImportDialog`

The final piece is adding the type to the component itself…

```ts
import React, { FunctionComponent } from 'react';

type Props = DispatchProps;

const ToolsPanel: FunctionComponent<Props> = ( … ) => …
```

Reply:
Thanks! I couldn't tell what this... WAS. Or is. There are a number of these kicking around the codebase that aren't defined as either Components or FunctionComponents. They normally have a signature like this one and a PropTypes definition at the end.